visualization/app/codeCharta/state/settings.service.ts

```diff
@@ -13,41 +13,39 @@ export interface SettingsServiceSubscriber {
 
 export class SettingsService implements FileStateServiceSubscriber {
 	private static SETTINGS_CHANGED_EVENT = "settings-changed"
+	private static DEBOUNCE_TIME = 400
 
 	private settings: Settings
-	private readonly debounceBroadcast: (update: RecursivePartial<Settings>) => void
+	private update: RecursivePartial<Settings> = {}
+	private readonly debounceBroadcast: () => void
 
 	constructor(private $rootScope: IRootScopeService, private $timeout: ITimeoutService, private loadingGifService: LoadingGifService) {
 		this.settings = this.getDefaultSettings()
-		this.debounceBroadcast = _.debounce(
-			(update: RecursivePartial<Settings>) =>
-				this.$rootScope.$broadcast(SettingsService.SETTINGS_CHANGED_EVENT, { settings: this.settings, update: update }),
-			400
-		)
+		this.debounceBroadcast = _.debounce(() => {
+			this.$rootScope.$broadcast(SettingsService.SETTINGS_CHANGED_EVENT, { settings: this.settings, update: this.update })
+			this.update = {}
+		}, SettingsService.DEBOUNCE_TIME)
 		FileStateService.subscribe(this.$rootScope, this)
 	}
 
 	public onFileSelectionStatesChanged(fileStates: FileState[], event: angular.IAngularEvent) {
-		this.updateSettings({
-			dynamicSettings: this.getDefaultDynamicSettingsWithoutMetrics()
-		})
-		this.updateSettings({
-			fileSettings: this.getNewFileSettings(fileStates)
-		})
+		this.resetDynamicAndFileSettings(fileStates)
 	}
 
-	public onImportedFilesChanged(fileStates: FileState[], event: angular.IAngularEvent) {}
+	public onImportedFilesChanged(fileStates: FileState[], event: angular.IAngularEvent) {
+		this.resetDynamicAndFileSettings(fileStates)
```

Comment:
Is this necessary, as the `onFileSelectionStatesChanged`-event always gets called after new files were imported?

Reply:
If that is the case, no. removed.